spec/examples/record/record_set_spec.rb

```diff
@@ -811,6 +811,46 @@ def self.latest(count)
     end
   end
 
+  describe '#page_size' do
+    it 'should perform query with specified page_size' do
+      expect_query_with_options(/SELECT/, :page_size => 1) do
+        Post.page_size(1).to_a
+      end
+    end
+
+    it 'should return the number of records specified by page_size' do
+      expect(Post.page_size(2).to_a.length).to be(2)
+    end
+  end
+
+  describe '#next_paging_state' do
+    it 'should return the paging state of the result' do
+      expect(Post.page_size(1).next_paging_state).not_to eq(nil)
+    end
+  end
+
+  describe '#paging_state' do
+    let(:page_one) { Post.page_size(1) }
+    let(:page_two) { Post.page_size(1).paging_state(page_one.next_paging_state) }
+
+    it 'should perform query with specified paging_state' do
+      expect(page_one.next_paging_state).not_to eq(nil)
+      expect_query_with_options(/SELECT/, :paging_state => page_one.next_paging_state) do
+        page_two.to_a
+      end
+    end
+
+    it 'should return results starting after previous page' do
+      expect_query_with_options(/SELECT/, :paging_state => page_one.next_paging_state) do
+        expect(page_two.to_a.first).to eq(posts.to_a.second)
+      end
+    end
+
+    it 'should be nil when omitted' do
+      expect(Post.page_size(1).paging_state(nil).to_a).to eq(page_one.to_a)
+    end
+  end
+
```

Comment:
Can we have a more behavior oriented test that retrieves all the records over however many pages there are and checks that we got all the expected values? I see the value of the preceding tests from a development perspective but from a maintenance perspective they check quite a bit of implementation detail (eg, the names of the paging control options to the cequel driver) that isn't really important to clients of this class. If those implementation details change in the future. these tests should continue to pass if the over all behavior is preserved.